### DISCUSSION TITLE
FYI, on Mosquitto 2.0.11 (Debian/bullseye)

### DISCUSSION THREAD

**Discussion by ThePeePs** [poblabs/weewx-belchertown#777] [General]
Just an FYI in case anyone else runs into the same issue I did while working to get TLS setup for my website to have it publicly available, I ran into a crazy bug with mosquitto and FIreFox where it was unable to open the MQTT websocket.

After _alot_ of debugging, and googling to figure out what the heck was going on, come to find out there's a bug in libwebsockets that gets triggered when an HTTP/2 request is made (for me it **ONLY** happened with FF, and TLS).  

After reading though issue [#1211](https://github.com/eclipse/mosquitto/issues/1211) on mosquitto's repo was able to find the answer.

The TL; DR version is, use the latest build from the [mosquitto repo](https://mosquitto.org/download/), or compile it from source. 

Since I run Debian, I just added their [repo](https://mosquitto.org/blog/2013/01/mosquitto-debian-repository/) to apt, updated and restarted mosquitto.  After that, the websocket via TLS works perfectly.

Hope this helps someone out.

**Comment by poblabs** [comment]
Today I've been troubleshooting a few issues with my mqtt broker. One is related to an SSL permissions error, but once I got that solved I noticed today that this bug may be affecting me as well. I've so far spent a couple hours troubleshooting. 

What version is your broker after adding their repo? 

I use mqtt in docker, and the version I am using 2.0.14 but still seems like if i have an empty username/password it cannot connect. Or there's something else going on

**Comment by poblabs** [comment]
Actually, think I just fixed it - at least for me - in this commit https://github.com/poblabs/weewx-belchertown/commit/f9763c5c6e21c790ddbb195071cf843b8c662f68

**Reply by ThePeePs** [reply]
I think i had put a fix for the blank username and password in my PR. I'm not at my computer right now, but iirc my mosquito version is 2.0.14.

**Reply by poblabs** [reply]
Oh shoot, I didn't see a PR for the mqtt fix - only your kiosk page.

**Reply by poblabs** [reply]
Also, 2.0.14 seems to be giving me problems in Docker. I ended up having to revert to 1.6.9.

**Reply by ThePeePs** [reply]
I dont play much with docker, I'm still "old school" with vm's. 
Also, just checked, confirmed version 2.0.14.

**Reply by ThePeePs** [reply]
> Oh shoot, I didn't see a PR for the mqtt fix - only your kiosk page.

No worries, I should have mentioned it when I opened the PR.
